### PR TITLE
Return valid HTTP 1.1 Status line

### PR DESCRIPTION
**Pull request by rdeutz**
Pull Request for Issue #16038

### Summary of Changes
Added more status to the responseMap and tried to get the best value for a code. Status should now a Valid HTTP 1.1 Status line as per https://www.w3.org/Protocols/rfc2616/rfc2616-sec6.html 

### Testing Instructions
Test different situations and check if the status is a valid status according to the document listet above.

### Before Patch
Not valid status codes

### After Patch
Valid status codes

### Documentation Changes Required
None

**Note: I had to change the unit test because they are testing wrong behaviour**

### CODE REVIEW

**Comment by brianteeman** [comment]
Can you order the status codes numerically please

**Comment by rdeutz** [comment]
@brianteeman done

File: libraries/joomla/application/web.php [hunk 1]
```diff
@@ -733,6 +785,27 @@ public function sendHeaders()
 	}
 
 	/**
+	 * Check is a given value can be successfully mapped to a valid http status value
```
Comment (brianteeman) [inline libraries/joomla/application/web.php]: Check if ...

**Comment by wilsonge** [comment]
The redirect function needs to now check if this is a 30x status code (the original point of the `$responseMap` class variable` rather than just checking if it is a key of that variable)

File: libraries/joomla/application/web.php [hunk 2]
```diff
@@ -660,10 +710,10 @@ public function setHeader($name, $value, $replace = false)
 		}
 
 		/**
-                 * If no keys found, safe to insert (!$keys)
-                 * If ($keys && $replace) it's a replacement and previous have been deleted
-                 * if($keys && !in_array...) it's a multiple value header
-                 */
+         * If no keys found, safe to insert (!$keys)
```
Comment (C-Lodder) [inline libraries/joomla/application/web.php]: indentation

File: libraries/joomla/application/web.php [hunk 3]
```diff
@@ -82,6 +82,19 @@ class JApplicationWeb extends JApplicationBase
...
+		200	=> 'HTTP/1.1 200 OK',
+		201	=> 'HTTP/1.1 201 Created',
+		202	=> 'HTTP/1.1 202 Accepted',
+		203	=> 'HTTP/1.1 203 Non-Authoritative Information',
+		204	=> 'HTTP/1.1 204 No Content',
+		205	=> 'HTTP/1.1 205 Reset Content',
+		206	=> 'HTTP/1.1 206 Partial Content',
+		207	=> 'HTTP/1.1 207 Multi-Status',
+		208	=> 'HTTP/1.1 208 Already Reported',
+		226	=> 'HTTP/1.1 226 IM Used',
```
Comment (C-Lodder) [inline libraries/joomla/application/web.php]: indentation

**Comment by rdeutz** [comment]
@wilsonge better?

File: libraries/joomla/application/web.php [hunk 4]
```diff
@@ -710,10 +726,10 @@ public function setHeader($name, $value, $replace = false)
 		}
 
 		/**
-         * If no keys found, safe to insert (!$keys)
-         * If ($keys && $replace) it's a replacement and previous have been deleted
-         * if($keys && !in_array...) it's a multiple value header
-         */
+		 * If no keys found, safe to insert (!$keys)
+		 * If ($keys && $replace) it's a replacement and previous have been deleted
+		 * if($keys && !in_array...) it's a multiple value header
```
Comment (Quy) [inline libraries/joomla/application/web.php]: Capitalize `If` and add a space after it.

File: libraries/joomla/application/web.php (hunk 3 above)
Comment (brianteeman) [inline libraries/joomla/application/web.php]: fixed

**Comment by wilsonge** [comment]
Looks good to me :)

**Comment by brianteeman** [comment]
@rdeutz can you resolve the conflicts please

**Comment by rdeutz** [comment]
fix and merged, @mbabker can you double check. Was a bit tricky to fix a conflict for a moved file.

**Pull request merged**